README.md

```diff
@@ -30,165 +32,471 @@ The two guiding principles are:
+| `browsers`  | `Array<String>`      | `['defaults']`   | [`last 1 Chrome versions`]                                         | An array of browserslist queries to specify targets for which to transpile/polyfill.         |
+| `node`      | `String`             | `current`        | `8.10`                                                             | A Node.js version identifier or `current` to specify for which target to transpile/polyfill. |
+
+#### Placeholders
+
+Hops supports a special syntax for placeholders. You can use angle brackets to reference other settings values:
+
+**`package.json`**
+
+```json
+{
+  "hops": {
+    "basePath": "/foo",
+    "assetPath": "<basePath>/assets"
+  }
+}
+```
+
+In the above example the value of `<basePath>` inside `assetPath` will be replaced with `/foo`, so `assetPath` becomes `/foo/assets`.  
+In case of nested objects you can also use the dot notation to access nested values:
+
+**`package.json`**
+
+```json
+{
+  "hops": {
+    "foo": {
+      "bar": "baz"
+    },
+    "qux": "<foo.bar>"
+  }
+}
 ```
 
-And write your main entry file (which should either be named `index.js` or should be referenced in your `package.json`s `main` field):
+#### Environment variables
+
+Hops has a concept of "universal environment variables", which are environment variables that are evaluated at _runtime_ instead of at _build time_.
+
+Usually when you reference `process.env.FOO` in your code, webpack will replace that expression with the value that the environment variable `FOO` had at build time. This is also true for Hops applications.
+
+**But:** Since Hops renders applications on the server- and client-side, we can reference environment variables in the settings and pass them on to the client and therefore allow users to have environment variables that are evaluated at runtime.
 
 **`package.json`**
 
 ```json
 {
-  "main": "main.js"
+  "hops": {
+    "myApiUrl": "[MY_API_URL]"
+  }
 }
 ```
 
-**`main.js`**
+The above example demonstrates how you can pass the value of the environment variable named `MY_API_URL` to Hops and later on you can access it in your application through its key like this:
+
+**`src/app.js`**
 
 ```javascript
-import { render } from 'hops-react';
+import { render, withConfig } from 'hops';
 import React from 'react';
 
-export default render(<h1>Hello World!</h1>);
+const Config = withConfig(({ config }) => <div>{config.myApiUrl}</div>);
+
+export default render(<Config />);
 ```
 
-For more details and advanced use-cases read the [full readme of `hops-react`](https://github.com/xing/hops/tree/master/packages/react).
+This allows you to have just one build artifact that you can use in different environments (staging, production, etc).
+
+**Caution:** Be careful not to expose secrets to the client side, because all environment variables that you reference this way in the settings will be visible in the rendered HTML.
```

Comment:
would it make sense to tell the people to use `process.env.*` on the server-side instead of Hops config when dealing with sensitive keys?

Reply:
I don't think we should mention that here (in the main readme, or in this section) because that only works in server mixins and the readers, as of yet, have never heard about mixins, nor should they.